crates/bevy_asset/src/assets.rs

```diff
@@ -443,6 +443,57 @@ macro_rules! load_internal_asset {
     }};
 }
 
+/// Loads an internal asset with its path.
+///
+/// Internal assets (e.g. shaders) are bundled directly into the app and can't be hot reloaded
+/// using the conventional API. See `DebugAssetServerPlugin`.
+#[macro_export]
+macro_rules! load_internal_asset_with_path {
```

Comment:
I don't understand the naming here. The `load_internal_asset` also takes in a path.

Also, it kinda feels like the `with_params` variant could be combined into one macro.

Reply:
right, it's not a great name. the distinction is that the path itself is passed to the asset constructor as well (so that internal shaders can reference their source files for error reporting). i'm very happy to take alternative names.

Reply:
How about modifying load_internal_asset to always pass a path to the asset constructor? Existing assets can just ignore it.

Reply:
> How about modifying load_internal_asset to always pass a path to the asset constructor? Existing assets can just ignore it.

this is a good call. i didn't realise but there's actually zero instances of `load_internal_asset` except for shader handles anyway.